sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala

```diff
@@ -219,6 +219,311 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     GenerateUnsafeProjection.generate(StartsWith(Literal("\"quote"), Literal("\"quote")) :: Nil)
   }
 
+  test("Support contains string expression with Collation") {
+    // Test 'contains' with different collations
+
+    // UCS_BASIC (default)
+    checkEvaluation(Contains(Literal(""), Literal("")), true)
```

Comment:
i would rather test these using the `sql` method

Reply:
StringExpressions are usually tested like this in `StringExpressionsSuite.scala`. They seem to be very handy for initial debugging and generally upgrading collation support in string expressions

On the other hand, I added a bunch of sql tests in `CollationSuite.scala` too! While I think it would be good to keep both for now, perhaps we can toss one of these suites away in the future?